modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/TransactionsInMultipleThreads.java

```diff
@@ -0,0 +1,481 @@
+import org.apache.ignite.internal.processors.cache.GridCacheAbstractSelfTest;
+import org.apache.ignite.internal.util.lang.GridAbsPredicate;
+import org.apache.ignite.lang.IgniteClosure;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.transactions.Transaction;
+import org.apache.ignite.transactions.TransactionConcurrency;
+import org.apache.ignite.transactions.TransactionIsolation;
+import org.apache.ignite.transactions.TransactionState;
+import org.junit.Assert;
+
+public class TransactionsInMultipleThreads extends GridCacheAbstractSelfTest {
+
+    /** Transaction concurrency control. */
+    private TransactionConcurrency transactionConcurrency;
+
+    /** Transaction isolation level. */
+    private TransactionIsolation transactionIsolation;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override protected void afterTestsStopped() throws Exception {
+        super.afterTestsStopped();
+        stopAllGrids();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override protected int gridCount() {
+        return 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override protected CacheMode cacheMode() {
+        return CacheMode.PARTITIONED;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override protected CacheAtomicityMode atomicityMode() {
+        return CacheAtomicityMode.TRANSACTIONAL;
+    }
+
+    /**
+     * Starts test scenario for all transaction controls, and isolation levels.
+     *
+     * @param consumer Scenario
+     */
+    private void withAllIsolationsAndConcurrencies(IgniteClosure<Object, Void> consumer) {
+        withAllIsolationsAndConcurrencies(consumer, null);
+    }
+
+    /**
+     * Starts test scenario for all transaction controls, and isolation levels.
+     *
+     * @param consumer Scenario
+     * @param arg
+     */
+    private void withAllIsolationsAndConcurrencies(IgniteClosure<Object, Void> consumer, Object arg) {
+        for (TransactionConcurrency concurrency : TransactionConcurrency.values()) {
+
+            for (TransactionIsolation isolation : TransactionIsolation.values()) {
+                this.transactionConcurrency = concurrency;
+                this.transactionIsolation = isolation;
+                try {
+                    consumer.apply(arg);
+                }
+                catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+
+        }
+    }
+
+    /**
+     * @throws IgniteCheckedException
+     */
+    public void testSimpleTransactionInAnotherThread() throws IgniteCheckedException {
+        withAllIsolationsAndConcurrencies(new IgniteClosure<Object, Void>() {
+            @Override public Void apply(Object o) {
+                try {
+                    TransactionsInMultipleThreads.this.simpleTransactionInAnotherThread();
+                }
+                catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
If you need to write it to log, please use Logger, otherwise no-operation requered.